internal/controllers/operator_controller_test.go

```diff
@@ -1018,6 +1023,94 @@ var _ = Describe("Operator Controller Test", func() {
 			Expect(cond.Message).To(Equal("installation has not been attempted as spec is invalid"))
 		})
 	})
+	When("a catalog changes on cluster", func() {
+		var testLogs, opNames []string
+		var cancel context.CancelFunc
+		var logCount int
+		BeforeEach(func() {
+			l := funcr.New(func(prefix, args string) {
+				if prefix == "operator-controller" &&
+					strings.Contains(args, `"controller"="operator"`) &&
+					strings.Contains(args, `"msg"="ending"`) {
+					// filter for only relevant logs
+					testLogs = append(testLogs, fmt.Sprintf("%s", args))
+				}
+			}, funcr.Options{Verbosity: 1})
```

Comment:
TIL, this `funcr` is super cool.

I'm torn though. On one hand, doing things this way means we can pretty much test our controller setup logic as-is. However it seems very brittle because changing our logs makes our tests for a completely unrelated thing break.

Another option is to slightly restructure the `SetupWithManager` like this:
```go
// SetupWithManager sets up the controller with the Manager.
func (r *OperatorReconciler) SetupWithManager(mgr ctrl.Manager) error {
	_, err := NewOperatorController(mgr, r)
	return err
}

// NewOperatorController sets up a controller for Operator objects using the provided reconciler.
func NewOperatorController(mgr ctrl.Manager, r reconcile.Reconciler) (controller.Controller, error) {
	return ctrl.NewControllerManagedBy(mgr).
		For(&operatorsv1alpha1.Operator{}).
		Watches(source.NewKindWithCache(&catalogd.Catalog{}, mgr.GetCache()),
			handler.EnqueueRequestsFromMapFunc(operatorRequestsForCatalog(context.TODO(), mgr.GetClient(), mgr.GetLogger()))).
		Owns(&rukpakv1alpha1.BundleDeployment{}).
		Build(r)
}
```

And then our tests could use `NewOperatorController` with a custom reconciler that collects the `reconcile.Requests` and makes assertions based on that. WDYT?